src/ui/char_selector.py

```diff
@@ -0,0 +1,47 @@
+from config import Config
+from utils.custom_mouse import mouse
+from utils.misc import cut_roi, wait
+from screen import Screen
+from template_finder import TemplateFinder
+
+
+class CharSelector:
+    def __init__(self, screen: Screen, config: Config):
+        self._config = config
+        self._char_template = None
+        self._screen = screen
+        self._template_finder = TemplateFinder(screen)
+
+    def has_char_template_saved(self):
+        return self._char_template is not None
+
+    def save_char_template(self):
+        img = self._screen.grab()
+        self._char_template = cut_roi(img, self._config.ui_roi["char_selection_top"])
+
+    def select_char(self):
+        if self._char_template is not None:
+            template_result = self._template_finder.search(
+                self._char_template,
+                self._screen.grab(),
+                threshold=0.8,
+                roi=self._config.ui_roi["char_selection_all"], normalize_monitor=True
+            )
+            if template_result.valid:
```

Comment:
maybe here would be wise to scroll down if the character is not visible upon restart, so to have something like

```
scrolls=0
template_result = # init to non valid somehow
while not template_result.valid and scrolls < 3:
  scrolls+=1
  mouse.wheel(-14) # in my system 14 scrolls down basically almost refresh the characters and you would reach the max character within 3 char scrolls I believe
  template_result =...
```
that would also require a change in custom mouse of the following function:
```
    @staticmethod
    def wheel(delta):
        _mouse.wheel(delta)
```
To be fair I haven't tested yet if the delta of the mouse actually correspond to 1 tick of my mouse so that's an assumption for now.

Reply:
Good point. Will test tomrrow.

Reply:
I've made the changes, please have a look, the only thing that is missing is identifying which character is currently selected rather than assuming the top character is the correct one

Reply:
by the way I have the changes locally to identify the character with a 0.99 match I just need to clean up a bit the code, since it's late I'll probably do it tomorrow

Reply:
Added the remaining changes, I added them as a pull request to this branch, rather than pushing directly to this branch as I did earlier. 
This way you can decide whether you want the changes or not